Makefile

```diff
@@ -1,11 +1,19 @@
 // TODO: add commands for build and run in dev/produciton mode
 // TODO: add commands for build protobuf files
-// TODO: add commands for lint/format/test project
 
 ROOT=$(realpath $(dir $(lastword $(MAKEFILE_LIST))))
 
-
-// TODO: add lint and format to github ci
 lint:
 	which golangci-lint || (go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.54.0)
-	golangci-lint run --config=$(ROOT)/.golangci.yml $(ROOT)/...
\ No newline at end of file
+	golangci-lint run --config=$(ROOT)/.golangci.yml $(ROOT)/...
+
+test:
+	go test -v ./...
+
+format:
+	@which gofumpt || (go install mvdan.cc/gofumpt@latest)
+	@gofumpt -l -w $(ROOT)
+	@which gci || (go install github.com/daixiang0/gci@latest)
+	@gci write $(ROOT)
+	@which golangci-lint || (go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.54.0)
+	@golangci-lint run --fix
```

Comment:
What `--fix` does here?

Reply:
golangci can auto fix some of the error that itself thrown